app/controllers/hostgroups_controller.rb

```diff
@@ -85,11 +85,11 @@ def puppetclass_parameters
   def environment_selected
     return not_found unless (@environment = Environment.find(params[:environment_id])) if params[:environment_id].to_i > 0
 
-    @hostgroup ||= Hostgroup.new
+    @hostgroup = Hostgroup.find_by_id(params[:hostgroup_id]) || Hostgroup.new
```

Comment:
I think the problem is that the JS that makes this call has to send `hostgroup_id` so that the Host form also understands it. If it had sent 'id', `@hostgroup` would be set correctly here, however it can't do that because it needs to tell the HostController that the `hostgroup_id` changed - and it can't do that by sending just 'id'. 

The patch works, it's either this or fix the JS to do different calls (not worth it in my opinion) - can you add a test though?

Thanks!